packages/ckeditor5-ui/src/dialog/dialog.ts

```diff
@@ -83,6 +83,7 @@ export default class Dialog extends Plugin {
 		// 'low' priority allows to add custom callback between `_hide()` and `onHide()`.
 		this.on<DialogHideEvent>( 'hide', () => {
 			this._onHide?.( this );
+			this._onHide = undefined;
```

Comment:
It's a nuance really. I was actually overeagerly cleaning the plugin state after the dialog is closed. But it wasn't necessary, because once the new dialog is shown, it overwrites the previous `_onHide` no matter if it is set or not - so it was set to `undefined` if no callback was specified and no older leftover was present. The only way to make it cause trouble would be firing `hide` event twice with no `show` event in between, which is not how the API should be used.

However after some thinking I came to the conclusion that it's a bit incoherent that we check `id` and set it only if present, but don't do the same with `onHide` - so I changed it, added proper unit test and left the line of code you commented in place.